homeassistant/helpers/network.py

```diff
@@ -45,6 +45,18 @@ def host_ip() -> str | None:
             )
         )
 
+    def addon_url() -> str | None:
+        if hass.config.api is None or not hass.components.hassio.is_hassio():
+            return None
+
+        return str(
+            yarl.URL.build(
+                scheme="http",
```

Comment:
scheme should by dynamic based on `hass.config.api.use_ssl`

Reply:
In that case let's add it to the if-check above and return None

Reply:
I added an option to the new helper to `allow_ssl`. It'll return you an https URL in this situation if set to `True` but it defaults to `False` (since most of the time that's no good).